models/migrations/v132.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/xorm"
+)
+
+func refixMergeBase(x *xorm.Engine) error {
+	type Repository struct {
+		ID        int64 `xorm:"pk autoincr"`
+		OwnerID   int64 `xorm:"UNIQUE(s) index"`
+		OwnerName string
+		LowerName string `xorm:"UNIQUE(s) INDEX NOT NULL"`
+		Name      string `xorm:"INDEX NOT NULL"`
+	}
+
+	type PullRequest struct {
+		ID         int64 `xorm:"pk autoincr"`
+		Index      int64
+		HeadRepoID int64 `xorm:"INDEX"`
+		BaseRepoID int64 `xorm:"INDEX"`
+		HeadBranch string
+		BaseBranch string
+		MergeBase  string `xorm:"VARCHAR(40)"`
+
+		HasMerged      bool   `xorm:"INDEX"`
+		MergedCommitID string `xorm:"VARCHAR(40)"`
+	}
+
+	var limit = setting.Database.IterateBufferSize
+	if limit <= 0 {
+		limit = 50
+	}
+
+	i := 0
+	for {
+		prs := make([]PullRequest, 0, 50)
+		if err := x.Limit(limit, i).Asc("id").Where("has_merged = ?", true).Find(&prs); err != nil {
+			return fmt.Errorf("Find: %v", err)
+		}
+		if len(prs) == 0 {
+			break
+		}
+
+		i += 50
```

Comment:
i += len(prs)

Reply:
@lunny what did you mean?

Reply:
as it may not be 50 that is added to `i` each time, for example if there are only 48 PRs in system, then I shouldn't add an additional 50 to count.

Reply:
Then the length of prs will equal 0.

i is not used in the loop

Reply:
I mean I'll change it but an SQL compliant DB will always return the full 50 results if it can. If it returns 48 then the that's the end of the results - it doesn't matter if you ask for the next 50 from that point, the point 2 places after or 2000 places after it.